float64 forward
float64 right
float64 down
float64 mforward
float64 mright
float64 mdown
